File: python/pe2.py
from os import system
system("cls")

def calculate_fib(lst):
    lst[0:2] = lst[1:]
    lst[2] = lst[0] + lst[1]
    

fib = [1, 2, 3]
sum = 2

while fib[2] < 4000000:
    calculate_fib(fib)
    if not(fib[2]%2):
        sum += fib[2]
print(sum)

File: cpp/02/pe2.cpp
#include <iostream>
#include <stdlib.h>

void calculate_fib(int fib[3]);

int main(){
    system("CLS");
    int fib[3] = {1, 2, 3}, sum=2;
    
    while(fib[2] < 4000000){
        calculate_fib(fib);
        sum += (!(fib[2]%2)) ? fib[2] : 0;
    }
    std::cout<<sum<<std::endl;
    return 0;
}

void calculate_fib(int fib[3]){
    fib[0] = fib[1];
    fib[1] = fib[2];
    fib[2] = fib[0] + fib[1];
}